Sampleapp

plivesey opened the pull request:
Adding a SampleApp.

Commit:
Initial commit for the sample app
Created app and Podfile. Ran pod install.

Commit:
Adding ChatsViewController
Switching from storyboards to nibs

Commit:
Adding models to the SampleApp

Commit:
Adding PINCache
Setting up RocketData
Fixing a bug in the docs

Commit:
Updating PINCache to get a working version
Adding stuff to get the Chats working

Commit:
Adding message view controller

Commit:
Adding fake push notifications so the users come off/online

Commit:
Adding some additional docs for the CacheDelegate and ParsingHelpers

Commit:
Adding some comments and new message push notifications

nicksnyder commented on SampleApp/SampleApp/ChatsViewController.swift:
```diff
@@ -0,0 +1,80 @@
+import UIKit
+import RocketData
+
+class ChatsViewController: UIViewController, CollectionDataProviderDelegate, UITableViewDataSource, UITableViewDelegate {
```
brief class doc to describe what this view controller does?

plivesey replied:
Fixed

nicksnyder commented on SampleApp/SampleApp/ChatsViewController.swift:
```diff
@@ -0,0 +1,80 @@
+class ChatsViewController: UIViewController, CollectionDataProviderDelegate, UITableViewDataSource, UITableViewDelegate {
+
+    let dataProvider = CollectionDataProvider<UserModel>()
+    let cacheKey = CollectionCacheKey.chat.cacheKey()
```
private?

plivesey replied:
Fixed and in other places too

nicksnyder commented on SampleApp/SampleApp/MessagesViewController.swift:
```diff
@@ -0,0 +1,104 @@
+import UIKit
+import RocketData
+
+class MessagesViewController: UIViewController, CollectionDataProviderDelegate, DataProviderDelegate, UITableViewDataSource, UITableViewDelegate {
```
I would add some class doc as well as inline doc in viewDidLoad to help readers understand what is happening

plivesey replied:
Fixed

nicksnyder commented on SampleApp/SampleApp/NetworkManager.swift:
```diff
@@ -0,0 +1,94 @@
+    static func fetchChats(completion: ([UserModel], NSError?)->Void) {
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) {
+            // Wait 1.5 seconds
+            usleep(UInt32(1.5 * 1000 * 1000))
```
don't want to use a single dispatch_after?

plivesey replied:
That would be too easy

plivesey replied:
Fixed

nicksnyder commented on SampleApp/SampleApp/RocketDataCacheDelegate.swift:
```diff
@@ -0,0 +1,88 @@
+    func setModel<T : SimpleModel>(model: T, forKey cacheKey: String, context: Any?) {
+        if let model = model as? SampleAppModel {
+            cache.setObject(model.data(), forKey: cacheKey, block: nil)
+        }
```
Do you want to show an example of handling the else case? (e.g. log or assert)

plivesey replied:
Fixed

nicksnyder commented:
lgtm

Commit:
added build images and inlined links in README

plivesey committed:
Fixing the race condition in tests that was causing travis-ci to fail. Fixes #11.

Fixing the race condition in tests that was causing travis-ci to fail. Fixes #11.

nicksnyder committed:
remove sentence fragment introduced in #13

Commit:
Adding a bunch of comments and logs to things to make it easier to read what's going on

Commit:
Adding private to a few variables

plivesey commented:
This is ready to go now @nicksnyder but found a very subtle bug which doing this. I'm going to fix that first, then merge this in.

plivesey commented:
#16 was merged in, so this is good to go now.
